### ISSUE TITLE
Build error ‘NWM_PLATFORM_LINUX’ was not declared

### ISSUE THREAD

**Issue by yuvadm** [fdb/frequensea#1]
Building on Arch x86_64 fails with this error:

```
Scanning dependencies of target frequensea
[ 76%] Building CXX object CMakeFiles/frequensea.dir/src/main.cpp.o
In file included from /home/yuval/dev/archlinux-packages/aur/frequensea-git/src/frequensea/src/main.cpp:14:0:
/home/yuval/dev/archlinux-packages/aur/frequensea-git/src/frequensea/src/main.cpp: In function ‘lua_State* l_init()’:
/home/yuval/dev/archlinux-packages/aur/frequensea-git/src/frequensea/src/nwm.h:46:26: error: ‘NWM_PLATFORM_LINUX’ was not declared in this scope
     #define NWM_PLATFORM NWM_PLATFORM_LINUX
                          ^
/home/yuval/dev/archlinux-packages/aur/frequensea-git/src/frequensea/src/main.cpp:1154:44: note: in expansion of macro ‘NWM_PLATFORM’
     l_register_constant(L, "NWM_PLATFORM", NWM_PLATFORM);
                                            ^
CMakeFiles/frequensea.dir/build.make:54: recipe for target 'CMakeFiles/frequensea.dir/src/main.cpp.o' failed
make[2]: *** [CMakeFiles/frequensea.dir/src/main.cpp.o] Error 1
CMakeFiles/Makefile2:60: recipe for target 'CMakeFiles/frequensea.dir/all' failed
make[1]: *** [CMakeFiles/frequensea.dir/all] Error 2
Makefile:76: recipe for target 'all' failed
make: *** [all] Error 2
```

What am I missing?

This is the `PKGBUILD` I'm working on if relevant: https://github.com/yuvadm/archlinux-packages/blob/master/aur/frequensea-git/PKGBUILD

**Comment by fdb** [comment]
HI,

The platform detection code in nwm.h has a bug. We didn’t test very much on Linux, I’m afraid (and we didn’t test *at all* on Windows). We’re integrating the code for the Raspberry Pi as well soon, but this fix should hopefully get you a bit further.


Best,

F

On 13 Mar 2015 at 10:31:41, Yuval Adam (notifications@github.com) wrote:

Building on Arch x86_64 fails with this error:

Scanning dependencies of target frequensea
[ 76%] Building CXX object CMakeFiles/frequensea.dir/src/main.cpp.o
In file included from /home/yuval/dev/archlinux-packages/aur/frequensea-git/src/frequensea/src/main.cpp:14:0:
/home/yuval/dev/archlinux-packages/aur/frequensea-git/src/frequensea/src/main.cpp: In function ‘lua_State* l_init()’:
/home/yuval/dev/archlinux-packages/aur/frequensea-git/src/frequensea/src/nwm.h:46:26: error: ‘NWM_PLATFORM_LINUX’ was not declared in this scope
     #define NWM_PLATFORM NWM_PLATFORM_LINUX
                          ^
/home/yuval/dev/archlinux-packages/aur/frequensea-git/src/frequensea/src/main.cpp:1154:44: note: in expansion of macro ‘NWM_PLATFORM’
     l_register_constant(L, "NWM_PLATFORM", NWM_PLATFORM);
                                            ^
CMakeFiles/frequensea.dir/build.make:54: recipe for target 'CMakeFiles/frequensea.dir/src/main.cpp.o' failed
make[2]: *** [CMakeFiles/frequensea.dir/src/main.cpp.o] Error 1
CMakeFiles/Makefile2:60: recipe for target 'CMakeFiles/frequensea.dir/all' failed
make[1]: *** [CMakeFiles/frequensea.dir/all] Error 2
Makefile:76: recipe for target 'all' failed
make: *** [all] Error 2

What am I missing?

This is the PKGBUILD I'm working on if relevant: https://github.com/yuvadm/archlinux-packages/blob/master/aur/frequensea-git/PKGBUILD

**Comment by yuvadm** [comment]
Fixed in b92672772dd4435fe84007b95f569086dc9193b0
